### PR TITLE
#ticket 33062 :sanitized filename as per windows filenaming conventions in Multiparser

### PR DESCRIPTION
**Pull request by vishalpandeyvip**
Ticket link : 
https://code.djangoproject.com/ticket/33062

This PR would sanitize filenames as per the naming convention for filenames in windows.
Changes are also made as suggested in PR #14834

The documentation which I have followed:
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions

### CODE REVIEW

File: django/http/multipartparser.py [hunk 1]
```diff
@@ -321,6 +321,23 @@ def sanitize_file_name(self, file_name):
...
+            'COM5', 'COM6', 'COM7', 'COM8', 'COM9', 'LPT1', 'LPT2', 
+            'LPT3', 'LPT4', 'LPT5', 'LPT6', 'LPT7', 'LPT8', 'LPT9' 
+        }
+
+        if file_name and file_name.split('.')[0].upper() in reserved_file_names:
+            file_name = file_name.replace(file_name.split('.')[0], '')
+        
+        extension = file_name.split('.')[-1]
+        for ext in ('..' + extension, ' .' + extension):
+            file_name = file_name.replace(ext, f'.{extension}')
```
Comment (felixxm) [inline django/http/multipartparser.py:339]: @vishalpandeyvip Have you missed my [comment](https://github.com/django/django/pull/14834#issuecomment-952762664)? IMO this goes too far and is unnecessarily complicated, we don't need `reserved_file_names` or hard-coded list of forbidden characters.

File: tests/file_uploads/tests.py [hunk 2]
```diff
@@ -38,6 +40,10 @@
     '../..\\hax0rd.txt',        # Relative path, mixed.
     '..&#x2F;hax0rd.txt',       # HTML entities.
     '..&sol;hax0rd.txt',        # HTML entities.
+    'hax><:?|*0rd.txt',         # with reserved characters.
+    'hax><:?|*0rd..txt',        # file name ends with a period.
```
Comment (felixxm) [inline tests/file_uploads/tests.py:44]: I'm also not sure why a file ended with `..txt` is an issue :thinking:

File: django/http/multipartparser.py (hunk 1 above)
Comment (vishalpandeyvip) [reply django/http/multipartparser.py:339]: I agree. I am removing these extra codes. I was trying to make changes as suggested in this [https://github.com/django/django/pull/14834#issuecomment-915974146](comment), In Microsoft documentation, there are some specific sets of reserved filenames and forbidden_characters, that's why I hardcoded them.

File: tests/file_uploads/tests.py (hunk 2 above)
Comment (vishalpandeyvip) [reply tests/file_uploads/tests.py:44]: From the documentation I mentioned above, it states "Do not end a file or directory name with a space or a period. Although the underlying file system may support such names, the Windows shell and user interface does not. However, it is acceptable to specify a period as the first character of a name. For example, ".temp"."
this was the reason behind adding those extra checks.

**Pull request closed without merging**